jobs/ror/ImportRorData.php

```diff
@@ -0,0 +1,168 @@
+                $jobs[] = new ProcessRorCsv(
+                    $pathCsv,
+                    $startRow,
+                    $endRow,
+                    $this->dataMapping,
+                    $this->dataMappingIndex,
+                    $this->noLocale,
+                    $this->temporaryTable,
+                    $this->scheduledTaskLogFilesPath
+                );
+            }
+
+            // we need to get thses in local variables to pass into the batch closures(e.g. then, catch, etc)
+            // as the serialization of $this in closure is buggy and can lead to issues
+            $logFilePath = $this->scheduledTaskLogFilesPath;
+
+        
+            UpdateRorRegistryDataset::log(
+                'RoR dataset importing batch process starting.', 
+                $logFilePath,
+                ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_NOTICE
+            );
+
+            Bus::batch($jobs)
+                ->then(function (Batch $batch) use ($pathCsv, $pathZipDir, $logFilePath) {
+                    UpdateRorRegistryDataset::cleanup([$pathCsv, $pathZipDir]);
+                    UpdateRorRegistryDataset::log(
+                        'RoR dataset importing completed.',
+                        $logFilePath,
+                        ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_COMPLETED
+                    );
+                })
+                ->catch(function (Batch $batch, Throwable $e) use ($logFilePath) {
+                    UpdateRorRegistryDataset::log(
+                        "RoR dataset importing batch at progress of {$batch->progress()}% have failed: {$e->getMessage()}",
+                        $logFilePath,
+                        ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_ERROR
+                    );
+                })
+                ->finally(function (Batch $batch) use ($pathCsv, $pathZipDir, $logFilePath) {
+                    UpdateRorRegistryDataset::cleanup([$pathCsv, $pathZipDir]);
+                    UpdateRorRegistryDataset::log(
+                        'RoR dataset importing batch jobs have finished.', 
+                        $logFilePath,
+                        ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_NOTICE
+                    );
+                })
+                ->name('import.ror.dataset.chunks')
+                ->dispatch();
+        } catch (Throwable $e) {
+            UpdateRorRegistryDataset::log(
+                "RoR dataset importing failed: {$e->getMessage()}",
+                $this->scheduledTaskLogFilesPath,
+                ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_ERROR
+            );
+
+            throw $e;
+        }
+    }
+
+    protected function countCsvRows(string $pathCsv): int
+    {
+        $rowCount = 0;
+        $handle = fopen($pathCsv, 'r');
+        while (!feof($handle)) {
+            $buffer = fread($handle, 8192);
+            $rowCount += substr_count($buffer, "\n");
```

Comment:
would it maybe be better to use `fgets` instead of `fread` here, or maybe even SplFileObject to get the total number of lines?